### PR TITLE
Refactor balance allowance

### PR DESCRIPTION
**Pull request by thepiwo**

### CODE REVIEW

File: contracts/fungible-token.aes [hunk 1]
```diff
@@ -52,68 +54,97 @@ contract FungibleToken =
...
-    Map.lookup_default((owner, spender), state.allowances, 0)
+  public function balance(owner: address) : option(int) =
+    Map.lookup(owner, state.balances)
+
+  public function allowance(allowance_accounts : allowance_accounts) : option(int) =
+    Map.lookup(allowance_accounts, state.allowances)
 
-  public function allowance_from(addr: address) : int =
-    Map.lookup_default((addr, Call.caller), state.allowances, 0)
+  public function allowance_for_caller(from_account: address) : option(int) =
```
Comment (mradkov) [inline contracts/fungible-token.aes:78]: Do you think the function naming here is clear enough?

**Review by mradkov — APPROVED** [review]

File: contracts/fungible-token.aes (hunk 1 above)
Comment (thepiwo) [reply contracts/fungible-token.aes:78]: I'm not sure, better than initially, do you have a suggestion on what to improve?

Comment (mradkov) [reply contracts/fungible-token.aes:78]: Maybe its just me or that's how I'm used to it, but I think the initial one is more clear. `allowance_from(account: address)` - returns what allowance do you have from the provided address.

Comment (thepiwo) [reply contracts/fungible-token.aes:78]: I'll adjust this

**Pull request merged**